src/Costura/CallAttach.cs

```diff
@@ -4,12 +4,18 @@
 
 partial class ModuleWeaver
 {
-    void CallAttach()
+    void CallAttach(Configuration config)
     {
-        if (FindInitializeCalls())
-            return;
+        var initialized = FindInitializeCalls();
 
-        AddModuleInitializerCall();
+        if (config.LoadAtModuleInit)
```

Comment:
No, you could have both the ModuleInitializer and a `CosturaUtility.Initializer`. I've made the templates threadsafe so they can be called multiple times.

This would allow the same code to run on Windows (ModuleInitializer) and Mono (CosturaUtility.Initializer).